spec/client/ics-008-wasm-client/README.md

```diff
@@ -84,329 +111,457 @@ interface Height {
-  portIdentifier: Identifier,
-  channelIdentifier: Identifier,
-  channelEnd: ChannelEnd) {
-    codeHandle = clientState.codeHandle()
-    assert(codeHandle.verifyChannelState(clientState, height, prefix, proof, portIdentifier, channelIdentifier, channelEnd))
+  path: CommitmentPath,
+  value: []byte
+): Error {
+  // check that the client is at a sufficient height
+  assert(clientState.latestHeight >= height)
+
+  payload = VerifyMembershipMsg{
+    height,
+    delayTimePeriod,
+    delayBlockPeriod,
+    proof,
+    path,
+    value
+  }
+
+  // retrieve client identifier-prefixed store
+	clientStore = provableStore.prefixStore("clients/{clientIdentifier}")
+
+  // use underlying wasm contract to verify that <path, value> has been stored
+  result = callContract(clientStore, clientState, marshalJSON(payload))
+  return result.error
 }
 
-function verifyPacketCommitment(
+function verifyNonMembership(
   clientState: ClientState,
   height: Height,
-  prefix: CommitmentPrefix,
+  delayTimePeriod: uint64,
+  delayBlockPeriod: uint64,
   proof: CommitmentProof,
-  portIdentifier: Identifier,
-  channelIdentifier: Identifier,
-  sequence: uint64,
-  commitment: bytes) {
-    codeHandle = clientState.codeHandle()
-    assert(codeHandle.verifyPacketCommitment(clientState, height, prefix, proof, portIdentifier, channelIdentifier, sequence, commitment))
+  path: CommitmentPath
+): Error {
+  // check that the client is at a sufficient height
+  assert(clientState.latestHeight >= height)
+
+  payload = verifyNonMembershipMsg{
+    height,
+    delayTimePeriod,
+    delayBlockPeriod,
+    proof,
+    path
+  }
+
+  // retrieve client identifier-prefixed store
+	clientStore = provableStore.prefixStore("clients/{clientIdentifier}")
+
+  // use underlying wasm contract to verify that nothing has been stored at path
+  result = callContract(clientStore, clientState, marshalJSON(payload))
+  return result.error
 }
+```
+### Wasm Contract Interface
 
-function verifyPacketAcknowledgement(
-  clientState: ClientState,
-  height: Height,
-  prefix: CommitmentPrefix,
-  proof: CommitmentProof,
-  portIdentifier: Identifier,
-  channelIdentifier: Identifier,
-  sequence: uint64,
-  acknowledgement: bytes) {
-    codeHandle = clientState.codeHandle()
-    assert(codeHandle.verifyPacketAcknowledgement(clientState, height, prefix, proof, portportIdentifier, channelIdentifier, sequence, acknowledgement))
-}
+#### Interaction between Go and Wasm
 
-function verifyPacketReceiptAbsence(
-  clientState: ClientState,
-  height: Height,
-  prefix: CommitmentPrefix,
-  proof: CommitmentProof,
-  portIdentifier: Identifier,
-  channelIdentifier: Identifier,
-  sequence: uint64) {
-    codeHandle = clientState.codeHandle()
-    assert(codeHandle.verifyPacketReceiptAbsence(clientState, height, prefix, proof, portIdentifier, channelIdentifier, sequence))
-}
+When an instruction needs to be executed in Wasm code, functions are executed using a `wasmvm`.
```

Comment:
Should we include that this wasmvm is sandboxed from any other vms users might write to for other purposes?

Reply:
Added a comment